bom/pom.xml

```diff
@@ -15,12 +15,30 @@
 
   <name>Google Guice - Bill of Materials</name>
 
+  <properties>
+    <guice.version>4.0.1-SNAPSHOT</guice.version>
+    <guice-assistedinject.version>4.0.1-SNAPSHOT</guice-assistedinject.version>
+    <guice-dagger-adapter.versionn>4.0.1-SNAPSHOT</guice-dagger-adapter.versionn>
+    <guice-dagger-adapter.version>4.0.1-SNAPSHOT</guice-dagger-adapter.version>
+    <guice-grapher.version>4.0.1-SNAPSHOT</guice-grapher.version>
+    <guice-jmx.version>4.0.1-SNAPSHOT</guice-jmx.version>
+    <guice-jndi.version>4.0.1-SNAPSHOT</guice-jndi.version>
+    <guice-multibindings.version>4.0.1-SNAPSHOT</guice-multibindings.version>
+    <guice-persist.version>4.0.1-SNAPSHOT</guice-persist.version>
+    <guice-servlet.version>4.0.1-SNAPSHOT</guice-servlet.version>
+    <guice-spring.version>4.0.1-SNAPSHOT</guice-spring.version>
+    <guice-struts2.version>4.0.1-SNAPSHOT</guice-struts2.version>
+    <guice-testlib.version>4.0.1-SNAPSHOT</guice-testlib.version>
+    <guice-throwingproviders.version>4.0.1-SNAPSHOT</guice-throwingproviders.version>
+    <guava.version>18.0</guava.version>
```

Comment:
I find this use of version properties odd, normally version properties are used sparingly - for example when you have several related dependencies and want to apply the same version across them. Using them to support overriding any dependency version in child projects is IMHO an anti-pattern as it makes the build less deterministic.

This change also seems specific to using the BOM as a parent pom, which I'd discourage for the reasons above. It's much simpler to import the BOM and then override specific dependency versions using `<dependencyManagement>`.